crates/bevy_ecs/hecs/src/query.rs

```diff
@@ -240,6 +240,85 @@ impl<'a, T: Component> Fetch<'a> for FetchMut<T> {
     }
 }
 
+macro_rules! impl_or_query {
+    ( $( $T:ident ),+; $( $t:ident ),+ ) => {
+        impl<$( $T: Query ),+> Query for Or<($( $T ),+)> {
+            type Fetch = FetchOr<($( $T::Fetch ),+)>;
+        }
+
+        impl<'a, $( $T: Fetch<'a> ),+> Fetch<'a> for FetchOr<($( $T ),+)> {
+            type Item = ($( $T::Item ),+);
+
+            fn access(archetype: &Archetype) -> Option<Access> {
+                if false $(|| $T::access(archetype).is_none() )+ {
```

Comment:
This isn't the correct access logic.

We need to got through each $T::Access and take the "max" access, where Write > Read > Iter.

(i didnt catch this in the other pr, but it has a similar problem)

Reply:
Or put another way, if T1 has Iter access and T2 has Write access, and T3 has None, we want to return Some(Write)

Reply:
Right ! I got it backwards for the Option, and then I thought that Changed, Mutated and Added all give Read access, but since Or can be used on any query...